### ISSUE TITLE
YOLO live tracking

### ISSUE THREAD

**Issue by ilyii** [allegroai/clearml#781]
I am currently training YOLO with ClearML running in the background. I am wondering if it's possible to live track the metrics like recall, precision and map in the ClearML interface in the form of line curves or plots.

**Comment by thepycoder** [comment]
Hi @ilyii !

Thanks for using ClearML :) You should be able to easily report metrics like recall, precision and mAP in the interface. In ClearML, metrics are called scalars. You can read more about reporting them here:

https://clear.ml/docs/latest/docs/guides/reporting/scalar_reporting/
https://clear.ml/docs/latest/docs/fundamentals/logger

I assume that in your YOLO training code there will be some section of code where you actually calculate the metrics you're interested in, ideally every n epochs or iterations. It might look something like this:

```
from clearml import Task
task = Task.init(....)

for epoch in amount_of_epochs:
### ... training or testing loop iteration code ...

  # somewhere in the code, you should have access to these metrics
  recall = calc_recall(args)
  precision = calc_precision(args)
  map = calc_map(args)
  
  # log them to clearml!
  task.get_logger().report_scalar(title="Precision-Recall", series="Recall", iteration=epoch, value=recall)
  task.get_logger().report_scalar(title="Precision-Recall", series="Precision", iteration=epoch, value=precision)
  task.get_logger().report_scalar(title="mAP", series="mAP", iteration=epoch, value=map)
```

Each plot has a title, but can have multiple series. So in this case both precision and recall will be plotted on the same plot named "Precision-Recall" as 2 separate lines with amount of iterations or epochs on the x-axis. mAP has a different title with only 1 series so it will be a separate plot with only 1 line.

Hope that makes sense!

**Comment by thepycoder** [comment]
@ilyii 

Just so I can help you better, which repository are you using to train? Which version of YOLO? We have an integration with YOLOv5, which does live track these metrics, you might look at the code as an example? https://github.com/ultralytics/yolov5

**Comment by ilyii** [comment]
First of all thanks for the help. I was training YOLOv5s, but the scalars, plots and debug samples tab was completely empty...

**Comment by ilyii** [comment]
Yes, I was using this repository. 
cmd: python3 train.py --data custom_data.yaml --cfg yolov5s.yaml  --batch-size 64 --img-size 700 --epochs 150

**Comment by thepycoder** [comment]
It seems to be working for me without issues, I get live tracking of the metrics you describe.

When you go to your ClearML task, can you see any output in the "Console logs" section? Is the model actually training there? Do you get output like this?
![Screenshot_20220920_093254](https://user-images.githubusercontent.com/11781950/191195323-56b71992-0aee-47a7-a1aa-ebcba37fa9a6.png)

Does it work for you when you run training on the coco sample dataset?
```
python train.py --img 640 --batch 16 --epochs 3 --data coco128.yaml --weights yolov5s.pt
```
Can you see any output in ClearML after running this command?

If possible a screenshot of the ClearML output is always appreciated :)

**Comment by ilyii** [comment]
I think it seems to work for me now as well. It somehow worked in the third or fourth attempt, idk what changed. Thank you anyway!
